Jetchat/app/src/main/java/com/example/compose/jetchat/components/JetchatDrawer.kt

```diff
@@ -0,0 +1,173 @@
+@Composable
+private fun DrawerHeader() {
+    Row(modifier = Modifier.padding(16.dp), verticalAlignment = CenterVertically) {
+        Image(
+            vectorResource(id = R.drawable.ic_jetchat),
+            modifier = Modifier.preferredSize(24.dp)
+        )
+        Image(
+            vectorResource(id = R.drawable.jetchat_logo),
+            modifier = Modifier.padding(start = 8.dp)
+        )
+    }
+}
+@Composable
+private fun DrawerItemHeader(text: String) {
+    ProvideEmphasis(emphasis = AmbientEmphasisLevels.current.medium) {
+        Text(text, style = MaterialTheme.typography.caption, modifier = Modifier.padding(16.dp))
+    }
+}
+
+@Composable
+private fun ChatItem(text: String, selected: Boolean, onChatClicked: () -> Unit) {
+    val background = if (selected) {
+        Modifier
+            .background(MaterialTheme.colors.primary.copy(alpha = 0.08f))
+    } else {
+        Modifier
+    }
+    Row(
+        modifier = Modifier
+            .preferredHeight(48.dp)
+            .fillMaxWidth()
+            .padding(horizontal = 8.dp, vertical = 4.dp)
+            .then(background)
+            .clip(MaterialTheme.shapes.medium)
+            .clickable(onClick = onChatClicked),
+        verticalAlignment = CenterVertically
+    ) {
+        ProvideEmphasis(emphasis = AmbientEmphasisLevels.current.medium) {
+            val mediumEmphasisOnSurface = AmbientEmphasisLevels.current.medium
+                .applyEmphasis(MaterialTheme.colors.onSurface)
+            Icon(
+                vectorResource(id = R.drawable.ic_jetchat),
+                tint = if (selected) MaterialTheme.colors.primary else mediumEmphasisOnSurface,
+                modifier = Modifier.padding(8.dp)
+            )
+            Text(
+                text,
+                style = MaterialTheme.typography.body2,
+                color = if (selected) MaterialTheme.colors.primary else AmbientContentColor.current,
+                modifier = Modifier.padding(8.dp)
+            )
+        }
+    }
+}
+
+@Composable
+private fun ProfileItem(text: String, @DrawableRes profilePic: Int?, onProfileClicked: () -> Unit) {
+    Row(
+        modifier = Modifier
+            .preferredHeight(48.dp)
+            .fillMaxWidth()
+            .padding(horizontal = 8.dp, vertical = 4.dp)
+            .clip(MaterialTheme.shapes.medium)
+            .clickable(onClick = onProfileClicked),
+        verticalAlignment = CenterVertically
+    ) {
+        ProvideEmphasis(emphasis = AmbientEmphasisLevels.current.medium) {
+            if (profilePic != null) {
+                Image(
+                    imageResource(id = profilePic),
+                    modifier = Modifier
+                        .padding(8.dp)
+                        .preferredSize(24.dp)
+                        .clip(CircleShape),
+                    contentScale = ContentScale.Crop
+                )
+            } else {
+                Spacer(modifier = Modifier.preferredWidth(24.dp).padding(8.dp))
```

Comment:
This modifier is shared with the `Image` above. I'd assign it to a variable so that the logic is easier to follow